src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -851,7 +859,10 @@ public bool TakeMessageHeaders(SocketInput input, FrameRequestHeaders requestHea
                 while (!scan.IsEnd)
                 {
                     var beginName = scan;
-                    scan.Seek(ref _vectorColons, ref _vectorCRs);
+                    if (scan.Seek(ref _vectorColons, ref _vectorCRs) == -1)
+                    {
+                        return false;
```

Comment:
I don't think so - at least I was able to make it error out. If you seek when there's no input available, the iterator doesn't move forward. Then you call `Take()` and get whatever was there under the iterator.